web/src/components/atomic/IconButton.vue

```diff
@@ -14,61 +14,40 @@
     rel="noopener noreferrer"
   >
     <slot>
-      <Icon :name="icon" />
+      <Icon v-if="icon" :name="icon" />
     </slot>
   </a>
   <button v-else :disabled="disabled" class="icon-button" type="button" :title="title" :aria-label="title">
     <slot>
-      <Icon :name="icon" />
+      <Icon v-if="icon" :name="icon" />
     </slot>
     <div v-if="isLoading" class="absolute left-0 top-0 right-0 bottom-0 flex items-center justify-center">
       <Icon name="loading" class="animate-spin" />
     </div>
   </button>
 </template>
 
-<script lang="ts">
-import { defineComponent, PropType } from 'vue';
+<script lang="ts" setup>
 import { RouteLocationRaw } from 'vue-router';
 
 import Icon, { IconNames } from '~/components/atomic/Icon.vue';
 
-export default defineComponent({
-  name: 'IconButton',
-
-  components: { Icon },
-
-  props: {
-    icon: {
-      type: String as PropType<IconNames>,
-      default: '',
-    },
-
-    disabled: {
-      type: Boolean,
-      required: false,
-    },
-
-    to: {
-      type: [String, Object, null] as PropType<RouteLocationRaw | null>,
-      default: null,
-    },
-
-    isLoading: {
-      type: Boolean,
-    },
-
-    title: {
-      type: String,
-      required: true,
-    },
-
-    href: {
-      type: String,
-      default: '',
-    },
+withDefaults(
+  defineProps<{
+    icon: IconNames | null;
+    disabled?: boolean;
+    to: RouteLocationRaw | null;
+    isLoading?: boolean;
+    title: string;
+    href?: string;
+  }>(),
```

Comment:
```suggestion
  defineProps<{
    icon: IconNames | null;
    disabled?: boolean;
    to?: RouteLocationRaw;
    isLoading?: boolean;
    title?: string;
    href?: string;
  }>(),
```

probably works without defaults and null as well